cordova-lib/src/plugman/fetch.js

```diff
@@ -255,13 +235,7 @@ function checkID(expectedIdAndVersion, pinfo) {
     if (!expectedIdAndVersion) return;
 
     var parsedSpec = pluginSpec.parse(expectedIdAndVersion);
-
-    if (parsedSpec.id != pinfo.id) {
-        var alias = parsedSpec.scope ? null : pluginMappernto[parsedSpec.id] || pluginMapperotn[parsedSpec.id];
```

Comment:
for this section, can you get rid of lines 260-263 but keep line 262.

It should look like 
```
 if (parsedSpec.id != pinfo.id) {
    throw new Error('Expected plugin to have ID "' + parsedSpec.id + '" but got "' + pinfo.id + '".');
}
```

That way, you can also undo your changes to cordova-lib/spec-plugman/fetch.spec.js. We should keep those tests I think.